tests/unit/defrag.cpp

```diff
@@ -566,3 +581,23 @@ BOOST_AUTO_TEST_CASE(test_defrag_in_dir) {
 	// check that defrag succeeded
 	BOOST_REQUIRE_EQUAL(eblob_stat_get(backend->stat, EBLOB_GST_DATASORT_COMPLETION_STATUS), 0);
 }
+
+
+BOOST_AUTO_TEST_CASE(test_defrag_by_position) {
+	const size_t TOTAL_RECORDS = 10;
+	eblob_config_test_wrapper config_wrapper(true, EBLOB_LOG_INFO);
+	config_wrapper.config.records_in_blob = TOTAL_RECORDS;
+	config_wrapper.config.blob_flags |= EBLOB_ITERATE_FLAGS_BY_POSITION;
+
+	eblob_wrapper wrapper(config_wrapper.config);
+	std::vector<item_t> shadow_elems;
+	auto generator = make_uniform_item_generator(wrapper, 10, 10);
+	fill_eblob(wrapper, shadow_elems, generator, TOTAL_RECORDS);
+
+	eblob_base_ctl *first_bctl = container_of(wrapper.get()->bases.next, eblob_base_ctl, base_entry);
+	BOOST_REQUIRE_EQUAL(eblob_generate_sorted_index(wrapper.get(), first_bctl), 0);
```

Comment:
maybe we could use autosort flag here

Reply:
Autosort flags proceeds in defrag thread, but all threads are disabled in unit tests